hidimstat/noise_std.py

```diff
@@ -1,66 +1,259 @@
 import numpy as np
 from numpy.linalg import norm
-from sklearn.linear_model import LassoCV, LassoLarsCV
+from scipy.linalg import toeplitz, solve
+from sklearn.linear_model import LassoCV, MultiTaskLassoCV
+from sklearn.model_selection import KFold
 
 
-def reid(X, y, method="lars", tol=1e-4, max_iter=1e+3, n_jobs=1):
+def reid(X, y, eps=1e-2, tol=1e-4, max_iter=1e+4, n_jobs=1, seed=0):
     """Estimation of noise standard deviation using Reid procedure
 
     Parameters
     -----------
-    X : ndarray or scipy.sparse matrix, (n_samples, n_features)
-        Data
-    y : ndarray, shape (n_samples,) or (n_samples, n_targets)
-        Target. Will be cast to X's dtype if necessary
-    method : string, optional
-        The method for the CV-lasso: "lars" or "lasso"
-    tol : float, optional
+    X : ndarray or scipy.sparse matrix, shape (n_samples, n_features)
+        Data.
+
+    y : ndarray, shape (n_samples,)
+        Target.
+
+    eps: float, optional (default=1e-2)
+        Length of the cross-validation path.
+        eps=1e-2 means that alpha_min / alpha_max = 1e-2.
+
+    tol : float, optional (default=1e-4)
         The tolerance for the optimization: if the updates are
         smaller than ``tol``, the optimization code checks the
         dual gap for optimality and continues until it is smaller
         than ``tol``.
-    max_iter : int, optional
-        The maximum number of iterations
+
+    max_iter : int, optional (default=1e+4)
+        The maximum number of iterations.
+
     n_jobs : int or None, optional (default=1)
         Number of CPUs to use during the cross validation.
+
+    seed: int, optional (default=0)
+        Seed passed in the KFold object which is used to cross-validate
+        LassoCV. This seed controls the partitioning randomness.
+
+    Returns
+    -------
+    sigma_hat : float
+        Estimated noise standard deviation.
+
+    beta_hat : array, shape (n_features,)
+        Estimated parameter vector.
```

Comment:
Do you provide somewhere a reference to Reid's paper ?

Reply:
you need to add a References section and use rst footnotes as done in sklearn for example